### PR TITLE
[bitnami/redis-cluster] Remove dead code in statefulset template

### PR DESCRIPTION
**Pull request by mboutet**
**Description of the change**

Remove dead code in statefulset template.

**Benefits**

Cleaner code.

**Possible drawbacks**

None that I can think of.

**Applicable issues**

None.

**Additional information**

None.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by javsalgar** [comment]
Hi,

Thank you very much for the PR! Could you explain the dead code?

**Comment by mboutet** [comment]
1. `spec.strategy` is not even a valid property for a `StatefulSet` (see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#statefulsetspec-v1-apps)
2. `updateStrategy` does not exist as a top-level key of the values. The correct values key is `statefulset.updateStrategy` which is referenced in the template here:
https://github.com/bitnami/charts/blob/a1f50de7e444341f62f289f988c0db676670de9a/bitnami/redis-cluster/templates/redis-statefulset.yaml#L394-L403

In other words, the removed lines in the template were never used nor rendered and were furthermore invalid.

**Review by javsalgar — APPROVED** [review]
LGTM!
